open-company/open-company-interaction
# Remove Slack comment sync'ing functionality.

[Trello](https://trello.com/c/G4634BfL)

This is the master PR. Also requires:

- [Slack Router](https://github.com/open-company/open-company-slack-router/pull/16)
- [Infrastructure](https://github.com/belucid/open-company-infrastructure/pull/117)

This PR removes the sync'ing of Slack/Carrot comments in an auto-shared post. It removes receipt of public channel messages from Slack that made that possible. It also moves the Slack bot usage response from Interaction (odd place!) to Slack router where it belongs.

**To test (locally):**

*First* make sure you create a Carrot post with sync'd comments with `mainline` branches. To do this:

- Run ngrok
- Config the local bot to point to your own local ngrok
- Configure a Carrot section to auto-post to a Slack test channel
- Create a post in that channel
- Comment on the post in Carrot
- [x] See the thread created with the comment in Slack?
- Comment back in Slack
- [x] See the Carrot comment created in Carrot?
- OK, all set, we'll use this post to test below

Now, ready for the rest of the testing.

Test there's no Slack firehose:

- Type some Slack messages into one of the test Slack channels, do you see the messages echo to Slack router and then onto some of the other services that watch its SNS events (Storage, Interaction...)
- Go [here](https://api.slack.com/apps), then to the local bot, then to "Event Subscriptions"
- In "Subscribe to Workspace Events", remove the "message.channels" event
- In "Subscribe to Bot Events", remove the "message.channels" event
- Be sure to click the "Save Changes" button
- (You can check the checklist item in Trello now if you want!)
- Swap your services over to `no-comment-sync`
- Now type some Slack messages into one of the test Slack channels.
- [x] Do you NOT see any of those Slack messages echo'ing back to the Slack router and onto the other services that watch its SNS events (Storage, Interaction...)
- FYI: you can leave these turned off now, we don't need them any longer on local bot

Test no Slack/Carrot comment sync'ing:

- Configure a Carrot section to auto-post to a Slack test channel
- Create a post in that channel
- Comment on the post in Carrot
- [x] no thread created?
- Go to the existing thread that was sync'd between Carrot and Slack that you created above
- Type a new comment in the Slack thread
- [x] Your comment didn't make it into Carrot?
- Type a new top level comment in Carrot
- [x] Your comment didn't make it into Slack?
 
Regression test commenting:

- Leave a top-level comment on a post
- Reply to the comment
- Edit your reply
- Edit your original comment
- Delete your reply
- Delete the original comment
- [x] No funny business?

Test Slack usage response:

- "Talk" to the bot by sending it a message
- [x] Do you get the canned response?
- [x] Did the canned response come from Slack router directly w/o involving interaction services (watch the console output)?

**To deploy:**

- Ensure you are on `no-comment-sync` for Ansible or have merged it
- Go [here](https://api.slack.com/apps), then to the Prod bot (called just "Carrot"), then to "Event Subscriptions"
- In "Subscribe to Workspace Events", remove the "message.channels" event
- In "Subscribe to Bot Events", remove the "message.channels" event
- Be sure to click the "Save Changes" button
- (You can check the checklist item in Trello now if you want!)
- Deploy Slack Router
- Deploy Interaction
